### ISSUE TITLE
Don't escape underscores inside {% in markdown

### ISSUE THREAD

**Issue by Haroenv** [prettier/prettier#3695]
**Prettier 1.9.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEwCkACAllMAbAVwBM4B9AJzjwEMYsA3ODUnAZxmrxrulYHoO5AEac8AOgC2RDGgC+AHSggANCAgAHHlFbJQ1cuQgB3AAr6EOlNXoQsRFSGrtkAM06s4qoeWpgA1nAwAMrqvjgA5sgw5ASeIEQQYK7ucWxw5DAmPuES1Ml4HqoAVqwAHgBCPv6BQdQScAAyOHD5hWoEMOodAEytcaHkHuTIILnkfglGSqrq5DgwAOp2MAAWyAAcAAwzhh4LPuojs3BDjA6UAI4EWJRZ1Dl5SG4FcR4SWFExrxF4cACKBAg8D6qg4QiWRFWyG6oJ8WDwEQAwhAJLkRlBoC1VAQPAAVahCSzPDyyWRAA)
```sh
--parser markdown
```

**Input:**
```markdown
{% include_relative _installations/tarball.md %}

```

**Output:**
```markdown
{% include_relative \_installations/tarball.md %}

```

**Expected behavior:**

**Output:**
```markdown
{% include_relative _installations/tarball.md %}

```

It otherwise causes Jekyll to fail

**Comment by Haroenv** [comment]
workaround: 

```md

<!-- prettier-ignore -->
{% include_relative _installations/tarball.md %}
```

**Comment by azz** [comment]
We don't officially support Jekyll, @ikatyang know of a mdast plugin to parse liquid tags?

**Comment by ikatyang** [comment]
As far as I can tell, there is no such plugin. 😵

**Comment by j-f1** [comment]
There [are some Liquid parsers](https://runkit.com/j-f/5a57f0870648740012b6a044) written in JS, but they require any non-default tags and/or filters to be custom-defined, and of course there are things like `{% if ... %}...{% endif %}` which should be internally indented.

Something we could do for now is to preserve the contents of anything matching `{{...}}` or `{%...%}`.

**Comment by Haroenv** [comment]
For my use case, the ignoring of those contents would be good enough, it was only adding an extra \ before _, and thus escaping that genuine character

**Comment by hcmlopes** [comment]
I'm a web developer for a Shopify eCommerce site and I live inside of liquid, HTML Liquid, SASS Liquid and JS Liquid. I would like to work on support for the liquid tags.
Where do I start ?

**Comment by j-f1** [comment]
The Markdown parser and printer are in [`src/language-markdown`](https://github.com/prettier/prettier/tree/master/src/language-markdown), and we use [remark](https://github.com/remarkjs/remark) to do the actual parsing work. You’d probably have to [create a remark plugin](https://github.com/remarkjs/remark/blob/master/doc/plugins.md#creating-plugins), then import it into Prettier and add it to the list of plugins used:

https://github.com/prettier/prettier/blob/84c603623da590487fa13bc6ae3fce4e80f465af/src/language-markdown/parser-markdown.js#L24-L29

**Comment by macs91** [comment]
@hcmlopes do you have updates regarding this issue? We use liquid too, we're having the same issue

**Comment by nhoizey** [comment]
I'm using Liquid tags in the Markdown content of my Jekyll blog.

I have also this issue, and another one.

If I type this in my Markdown file:

```
{% cloudinary nice_prefix_-_for_the_filename.jpg %}
```

Prettier changes it to that:

```
{% cloudinary nice*prefix*-\_for_the_filename.jpg %}
```

There is the issue with the `\` added in front of one of the `_`, but there are also `_`s that are changed to `*`s.

**Comment by lydell** [comment]
@nhoizey Yeah, all sorts of unexpected things can happen inside `{%` since Prettier has no idea what that is – it just formats everything as regular text/markdown.

**Comment by suchipi** [comment]
We should be able to fix this by detecting paragraphs whose first and last words are the word `{%` and `%}` and printing those entire paragraphs as they appear in the source.

**Comment by nhoizey** [comment]
@lydell I understand it can be tricky.

@suchipi Liquid tags can be used inside paragraph, not only on isolated lines, AFAIK.

**Comment by suchipi** [comment]
@nhoizey are they used like handlebars? Where you can put strings wherever you want?

**Comment by j-f1** [comment]
Yes.

**Comment by suchipi** [comment]
Thanks @j-f1

**Comment by hcmlopes** [comment]
To answer @macs91 question, I have had no updates. My knowledge in this area is very limited and I haven't had too much time to learn any further.
I have definitely not forgotten about it though as I work with messy liquid files on a daily basis

**Comment by duailibe** [comment]
I'll give this a shot

**Comment by duailibe** [comment]
Also can anyone point me at Markdown documents with those tags so I can test a solution?

**Comment by j-f1** [comment]
> [Here is an example from a project I contribute to](https://github.com/Charcoal-SE/charcoal-jekyll-theme/blob/abbc3f27aa88bcbf5126e85c506beba03d732598/_layouts/default.html#L73-L91).

**Comment by duailibe** [comment]
I asked for Markdown documents. That doesn't help because we print HTML as-is

**Comment by j-f1** [comment]
@duailibe Sorry, I missed that part. [How about this one](https://raw.githubusercontent.com/Charcoal-SE/charcoal-se.github.io/site/scripts.md)?

**Comment by duailibe** [comment]
@j-f1 awesome! Thanks

**Comment by duailibe** [comment]
Opened a PR #4484, you can try it out on the playground in: https://deploy-preview-4484--prettier.netlify.com/playground/

It would be awesome to get some real code examples to test this :)

**Comment by Haroenv** [comment]
Thx @duailibe
